pkg/webhook/mutation/pod/init_container.go

```diff
@@ -45,6 +45,10 @@ func initContainerResources(dk dynakube.DynaKube) corev1.ResourceRequirements {
 	}
 
 	if !dk.OneAgent().IsCSIAvailable() {
+		if dk.MetadataEnrichmentEnabled() {
+			return defaultInitContainerResources()
+		}
+
```

Comment:
This is the:
no-csi + metadata only ==> default limit

what about the ?
no-csi + metadata + oa ==> no limit (unless custom is set)

```
	t.Run("should have no if OA is enabled and csi is disabled", func(t *testing.T) {
		installconfig.SetModulesOverride(t, installconfig.Modules{CSIDriver: false})

		dk := getTestDynakubeDefaultAppMon()

		dk.Spec.MetadataEnrichment.Enabled = ptr.To(true)
                dk.Spec.OneAgent.CloudNativeFullStack = &oneagent.CloudNativeFullStackSpec{}

		initResources := initContainerResources(*dk)

                require.Empty(t, initResources)
	})
```

Reply:
should be covered here https://github.com/Dynatrace/dynatrace-operator/pull/4288/commits/94f387c3dae92d16be3327b5ee15399b9c438de2